sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/models.py

```diff
@@ -493,18 +493,341 @@ def request_id(self):
+            san_emails=san_emails,
+            san_dns_names=san_dns_names,
+            san_upns=san_upns
+        )
+
+    def _to_x509_properties_bundle(self):
+        # type: (X509Properties) -> models.X509CertificateProperties
+        """Converts a X509Properties to an autorest-generated X509Properties"""
+        sans = models.SubjectAlternativeNames(
+            emails=self.san_emails,
+            dns_names=self.san_dns_names,
+            upns=self.san_upns
+        )
+        return models.X509CertificateProperties(
+            subject=self.subject_name,
+            ekus=self.ekus,
+            key_usage=self.key_usage.value if isinstance(self.key_usage, KeyUsageType) else self.key_usage,
+            validity_in_months=self.validity_in_months,
+            subject_alternative_names=sans
+        )
+
+    @property
+    def subject_name(self):
+        # type: () -> str
+        """:rtype: str"""
+        return self._subject_name
+
+    @property
+    def ekus(self):
+        # type: () -> list[str]
+        """The enhanced key usage.
+
+        :rtype: list[str]
+        """
+        return self._ekus
+
+    @property
+    def key_usage(self):
+        # type: () -> list[KeyUsageType]
+        """List of key usages.
+
+        :rtype: list[~azure.keyvault.certificates.enums.KeyUsageType]
+        """
+        return self._key_usage
+
+    @property
+    def validity_in_months(self):
+        # type: () -> int
+        """The duration that the certificate is valid for in months.
+
+        :rtype: int
+        """
+        return self._validity_in_months
+
+    @property
+    def san_emails(self):
+        # type: () -> list[str]
+        """The subject alternative email addresses.
+
+        :rtype: list[str]
+        """
+        return self._san_emails
+
+    @property
+    def san_dns_names(self):
+        # type: () -> list[str]
+        """The subject alternative domain names.
+
+        :rtype: list[str]
+        """
+        return self._san_dns_names
+
+    @property
+    def san_upns(self):
+        # type: () -> list[str]
+        """The subject alternative user principal names.
+
+        :rtype: list[str]
+        """
+        return self._san_upns
+
+
+class IssuerParameters(object):
+    """Parameters for the issuer of the X509 component of a certificate.
+
+    :param issuer_name: Name of the referenced issuer object or reserved names; for
+     example, 'Self' or 'Unknown'.
+    :type name: str
+    :param certificate_type: Type of certificate to be requested from the
+     issuer provider.
+    :type certificate_type: str
+    :param certificate_transparency: Indicates if the certificates generated
+     under this policy should be published to certificate transparency logs.
+    :type certificate_transparency: bool
+    """
+
+    def __init__(
+        self,
+        issuer_name,  # type: str
+        certificate_type=None,  # type: str
+        certificate_transparency=None  # type: bool
```

Comment:
Everything but `issuer_name` is `Optional`.

Reply:
thanks!